src/chains/ton/depositAndCall.ts

```diff
@@ -0,0 +1,100 @@
+import { Address, beginCell, toNano } from "@ton/core";
+import { stringToCell } from "@ton/core/dist/boc/utils/strings";
+import { KeyPair, mnemonicToWalletKey, mnemonicValidate } from "@ton/crypto";
+import { TonClient, WalletContractV4 } from "@ton/ton";
+import { Gateway } from "@zetachain/protocol-contracts-ton/dist/wrappers";
+import { AbiCoder, ethers } from "ethers";
+
+import { ParseAbiValuesReturnType } from "../../../types/parseAbiValues.types";
+import { getAddress } from "../../../utils/getAddress";
+
+type tonDepositAndCallParams = {
+  amount: string;
+  data?: string;
+  receiver: string;
+  types?: string[];
+  values?: ParseAbiValuesReturnType;
+};
+
+type tonOptions = {
+  apiKey?: string;
+  chainId: string;
+  gateway?: string;
+  keyPair?: KeyPair;
+  rpc: string;
+  signer?: string;
+  wallet?: WalletContractV4;
+};
+
+const getAccountFromMnemonic = async (mnemonicRaw: string) => {
+  // Remove extra spaces and newlines
+  const mnemonic = mnemonicRaw.trim().replace(/\s+/g, " ");
+
+  const mnemonicWords = mnemonic.split(" ");
+
+  if (!(await mnemonicValidate(mnemonicWords))) {
+    throw new Error("Invalid mnemonic phrase");
+  }
+
+  const keyPair = await mnemonicToWalletKey(mnemonicWords);
+
+  const wallet = WalletContractV4.create({
+    publicKey: keyPair.publicKey,
+    workchain: 0,
+  });
+
+  return { keyPair, wallet };
+};
```

Comment:
This function is repeated on both ton functions, it would be much more beneficial to extract and reuse.